packages/concerto-util/lib/baseexception.js

```diff
@@ -27,12 +30,14 @@ class BaseException extends Error {
      * Create the BaseException.
      * @param {string} message - The exception message.
      * @param {string} component - The optional component which throws this error.
+     * @param {string} errorType - The optional error code regarding the error
      */
-    constructor(message, component) {
+    constructor(message, component, errorType) {
```

Comment:
added a errortype parameter which is optional and provides the errorcode related to the error